### PR TITLE
Support gRPC + Protobuf in Rust

### PR DESCRIPTION
**Pull request by alexjpwalker**
## What is the goal of this PR?

We've added crates required to support gRPC and Protobuf in Rust.

## What are the changes implemented in this PR?

- Add `futures`, `grpc`, `protobuf` and `tls-api` to our Cargo Raze crates

### CODE REVIEW

File: builder/rust/deps.bzl [hunk 1]
```diff
@@ -3,10 +3,10 @@ load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive", "http_file"
 def deps():
     http_archive(
         name = "rules_rust",
-        sha256 = "531bdd470728b61ce41cf7604dc4f9a115983e455d46ac1d0c1632f613ab9fc3",
-        strip_prefix = "rules_rust-d8238877c0e552639d3e057aadd6bfcf37592408",
+        sha256 = "7240a4865b11427cc58cd00b3e89c805825bfd3cc4c225b7e992a58622bec859",
+        strip_prefix = "rules_rust-a619e1a30bb274639b6d2ccb76c820a02b9f94be",
         urls = [
-            "https://github.com/bazelbuild/rules_rust/archive/d8238877c0e552639d3e057aadd6bfcf37592408.tar.gz",
+            "https://github.com/bazelbuild/rules_rust/archive/a619e1a30bb274639b6d2ccb76c820a02b9f94be.tar.gz",
```
Comment (alexjpwalker) [inline builder/rust/deps.bzl:9]: Just updating to the latest `main` in `rules_rust`

File: distribution/deps.bzl [hunk 2]
```diff
@@ -21,5 +21,5 @@ def vaticle_bazel_distribution():
     git_repository(
         name = "vaticle_bazel_distribution",
         remote = "https://github.com/vaticle/bazel-distribution",
-        commit = "d3ce7e6a7787e6d1ecc22cf2f69a9952553066e7" # sync-marker: do not remove this comment, this is used for sync-dependencies by @vaticle_bazel_distribution
+        commit = "4d6d13878018601576a9dc0cec52d8f68fc0d97e" # sync-marker: do not remove this comment, this is used for sync-dependencies by @vaticle_bazel_distribution
```
Comment (alexjpwalker) [inline distribution/deps.bzl:24]: Updating to the latest `master`, no significant updates

File: library/crates/BUILD.bazel [hunk 3]
```diff
@@ -23,7 +23,79 @@ alias(
```
Comment (alexjpwalker) [inline library/crates/BUILD.bazel:23]: Adding `futures`, `grpc`, `protobuf` and `tls_api` crates required for `typedb-client-rust` and `typedb-protocol`

File: library/crates/Cargo.toml [hunk 4]
```diff
@@ -9,6 +9,16 @@ path = "" # ignored by cargo raze
 antlr-rust = "0.2.0"
```
Comment (alexjpwalker) [inline library/crates/Cargo.toml:9]: These dependencies and aliased targets are used by TypeDB Client Rust and the Protobuf compiler that TypeDB Protocol will use

File: library/ortools/rust/BUILD [hunk 5]
```diff
@@ -16,7 +16,7 @@
 #
 
 
-load("@rules_rust//rust:rust.bzl", "rust_binary", "rust_library")
+load("@rules_rust//rust:defs.bzl", "rust_library")
```
Comment (alexjpwalker) [inline library/ortools/rust/BUILD:19]: rust.bzl no longer exists in rules_rust.

File: library/crates/Cargo.toml [hunk 6]
```diff
@@ -9,6 +9,16 @@ path = "" # ignored by cargo raze
...
 rocksdb = "0.17.0"
 cxx = "1.0.59"
+futures = "0.3.21"
+
+# gRPC dependencies
```
Comment (alexjpwalker) [inline library/crates/Cargo.toml:14]: These dependencies, raze arguments and aliased targets are taken from the `rules_rust` Protobuf setup guide at https://bazelbuild.github.io/rules_rust/rust_proto.html.

The `Cargo.toml` file that `rules_rust` recommends using is located at https://github.com/bazelbuild/rules_rust/blob/f3a34f67e97a12aaf3adaa9cfa957dceb9ad7b89/proto/raze/Cargo.toml.

We've adapted it slightly for our needs, notably by pinning the `log` version to `0.4.6`.

**Review by lolski — APPROVED** [review]

**Pull request merged**